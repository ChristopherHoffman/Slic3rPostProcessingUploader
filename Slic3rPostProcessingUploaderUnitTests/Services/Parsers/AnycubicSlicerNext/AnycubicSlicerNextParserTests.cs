using Slic3rPostProcessingUploader.Services.Parsers.AnycubicSlicerNext;
using Snapshooter.MSTest;


namespace Slic3rPostProcessingUploaderUnitTests.Services.Parsers.AnycubicSlicerNext
{
    [TestClass]
    public class AnycubicSlicerNextParserTests
    {
        [TestMethod]
        public void ShouldReturnAnEmptySettingWhenGcodeIsEmpty()
        {
            var parser = new AnycubicSlicerNextParser("");
            var result = parser.ParseGcode("");
            Assert.IsNotNull(result);

            Snapshot.Match(result);
        }

        [TestMethod]
        public void ShouldReturnExpectedValuesWhenGivenFullGcode()
        {
            var parser = new AnycubicSlicerNextParser("");
            var result = parser.ParseGcode(AnycubicSlicerNextParserTestGcode.CalibrationCube);

            Snapshot.Match(result);
        }

        [TestMethod]
        public void ShouldRenderTheExpectedNoteWhenGivenATemplateWithNoReplacements()
        {
            string template = "Settings:";

            var parser = new AnycubicSlicerNextParser(template);
            var result = parser.ParseGcode(AnycubicSlicerNextParserTestGcode.CalibrationCube);

            Assert.AreEqual("Settings:", result.settings.note);
        }

        [TestMethod]
        public void ShouldRenderTheExpectedNoteWhenGivenATemplateWithASingleReplacement()
        {
            string template = """
                Settings:
                    Layer Height: {{layer_height}}
                """;

            var parser = new AnycubicSlicerNextParser(template);
            var result = parser.ParseGcode(AnycubicSlicerNextParserTestGcode.CalibrationCube);

            Assert.AreEqual("""
                Settings:
                    Layer Height: 0.2
                """, result.settings.note);
        }

        [TestMethod]
        public void ShouldRenderTheExpectedNoteWhenGivenATemplateWithMultipleReplacements()
        {
            string template = """
                Settings:
                    Layer Height: {{layer_height}}
                    First Layer Height: {{first_layer_height}}
                    Wall Loops: {{wall_loops}}
                    Top Shell Layers: {{top_shell_layers}}
                    Bottom Shell Layers: {{bottom_shell_layers}}
                    Sparse Infill Density: {{sparse_infill_density}}
                """;

            var parser = new AnycubicSlicerNextParser(template);
            var result = parser.ParseGcode(AnycubicSlicerNextParserTestGcode.CalibrationCube);

            Assert.AreEqual("""
                Settings:
                    Layer Height: 0.2
                    First Layer Height: 0.200
                    Wall Loops: 2
                    Top Shell Layers: 5
                    Bottom Shell Layers: 3
                    Sparse Infill Density: 15%
                """, result.settings.note);
        }

        [TestMethod]
        public void ShouldRenderFullTemplateWhenGivenAGcodeWithTwoFilaments()
        {
            var parser = new AnycubicSlicerNextParser("");
            var result = parser.ParseGcode(AnycubicSlicerNextParserTestGcode.TwoFilamentCalibrationCube);

            Snapshot.Match(result);
        }

    }
}
