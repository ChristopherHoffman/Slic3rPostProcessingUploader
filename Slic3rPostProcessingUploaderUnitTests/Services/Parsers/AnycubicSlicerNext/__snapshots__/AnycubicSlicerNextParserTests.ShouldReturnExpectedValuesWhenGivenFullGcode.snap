{
  "Slicer": "AnycubicSlicerNext",
  "CuraVersion": "1.3.2",
  "PluginVersion": null,
  "settings": {
    "note": "Settings:\n\nQuality:\n  Layer Height:\n    Layer Height: 0.2\n    First Layer Height: 0.200\n  Line Width:\n    Default Line Width: 0.42\n  Seam:\n    Seam Position: aligned\n  Precision:\n    Precise Wall: 0\n  Ironing:\n    Ironing: \n  Wall Generator:\n    Generator: classic\n  Walls And Surfaces:\n    Only One Wall on Top Surfaces: 1\n\nStrength:\n  Walls:\n    Wall Loops: 2\n  Top/Bottom Shells:\n    Top Shell Layers: 5\n    Top Shell Thickness: 1\n    Bottom Shell Layers: 3\n    Bottom Shell Thickness: 0.6\n  Infill:\n    Sparse Infill Density: 15%\n    Sparse Infill Pattern: gyroid\n\nSpeed:\n  First Layer Speed:\n    First Layer: 45\n    Number of Slow Layers: 1\n  Other Layers Speed:\n    Outer Wall: 130\n    Inner Wall: 200\n    Sparse Infill: 300\n    Internal Solid Infill: 130\n    Top Surface: 130\n  Travel Speed:\n    Travel: 350\n\nSupport:\n  Support:\n    Enable Support: 0\n    Type: tree(auto)\n    Style: default\n    Threshold Angle: 40\n    On Build Plate Only: 1\n  Raft:\n    Raft Layers: 0\n\nMultimaterial:\n  Prime Tower:\n    Enable: 0\n    Width: 35\n    Brim Width: 5\n  Ooze Prevention:\n    Enable: 0\n  Flush Options:\n    Flush into Objects' Infill: 0\n    Flush into Objects' Support: 1\n  Advanced:\n    Use Beam Interlocking: 0\n\nOther:\n  Skirt:\n    Skirt Type: combined\n    Skirt Loops: 0\n    Skirt Min Extrusion Length: 0\n  Brim:\n    Brim Type: auto_brim\n    Brim Width: 5\n  Special Mode:\n    Slicing Mode: regular\n    Print Sequence: by layer\n    Spiral Vase: 0\n    Fuzzy Skin: none",
    "print_name": null,
    "time_stamp": null,
    "estimated_print_time_seconds": 1121,
    "material_used_mg": 3530,
    "Snapshot": "iVBORw0KGgoAAAANSUhEUgAAAOYAAABuCAYAAAAziW8OAAAEUklEQVR4Ae3bPYtcVRjA8fsB7GwsFB\nSCIFoYU0gUJawivq1vEI2CEkGIjaCIJNpkC0VEGxGxXC1tRLATYSstFazsFmzzISbcJBcmy87szD3n\n3vOce38/eCCk3j/P4ZmZpgEAmJW77n/8XAPE8PALn+y9+8O1RTvtvxugrPOf/33YRdlN+3+2JxSwvC\nVXje0JI9okym6e++jXgwYYTvs8Pe7petJ42sJAttmSq+aJd77bb4A8+mxJ2xMGkmNLOgxBRu3RZqgo\nlw9Dd9x5z70NsN6QW9LTFnoYO8qj27MBbpfzwGN7QqKSW3LVOAwxaxG25Lrt6TDErETckrYns1ZTlN\n04DDFZfb/nGmUchpicGrfkqvG0ZRJq3pK2J5MzpS1pezIJ7c+ragosZRyGCO/U2QsXa4oq13jaEtYc\nnq4njR9iE8oUDzx9x/akOFty9TgMUcQYP2SufXzfltHYktuP7cmgRNl/fKzCIBx40sdhiGxsyfzjaU\nuS0y9e3q/pD76mOfPyZz7zpJ8Hzr23/8wHPy/e/Pq/qv7oI88bX/67ePLi94tTj74uTPppw9y59OOi\nnZc+/aOqACLO7uXfF2cvfHVjhElvy2G2Y3v2n3ZLdlEKkyRHw+zm1at/VhVFyVneksIki1VhtvP8x7\n8t3v72/6oiGXueev+nY6MUJknWhdnN+S/+qSqWMWbVlhQmWWwSZrc9awqndJTCJMmmYToM3ZyjBx5h\nMohtwpzzYWjTLSlMsugTZrc953IY2mZLCpMs+oY5h+3ZZ0sKkyxSw5zqYSg1SmGSJEeYUzoMdd9zTY\n1SmCTJFeYUnrY5tqQwySJ3mLVuz1xbUphkMUSYNW3P3FtSmGQxZJjRD0PrvucqTIoaOsyIT9sht6Qw\nyWKMMLuJ8EPssaIUJknGDLP09hziwCNMBjF2mDsFDkNjbklhkkWpMLvtOfT3bcfeksIki5JhDrk9S2\n1JYZJFhDDbyfmxSoQohUmSKGF2T9uUw1DO77kKk6IihZnytI2yJYVJFhHD3HZ7RtqSwiSLqGFusj0j\nbklhkkX0MNs57jD07Ie/hI5SmCSpIczlp+1rV/8K+3QVJtnUEmY3NQQpTJIJU5gEJExhEpAwhUlAwh\nQmAQlTmAQkTGESkDCFSUDCFCYBCVOYBCRMYRKQMIVJQMIUJkHdd+aVPWHmnbsfenqvgRyEmT6P7F45\nbCC307tXDoTp6UpQwrQlCSri9hQlNPEOQ5GCdOChuMfe+uZQmLYkAUXYnqWjtCUJa45h2pJUodRhSJ\nSwgSmHKUiqNub2FCVsYazD0BhROvAwOTWHaUsyaUM+bYeK8sGdSwcNzEENYdqSzFLu7SlKyCTnYShX\nkA48cEuO79vakjCA1KdtSpS2JJxgzDBtSdhCn+0pShjJEGEKEjLY9DAkShjZJh+rOPBAIX3CtCVhBK\nsOQ8dF6XuuMLJ1YdqSUNDy9hQlBNIdhhx4AACAwq4DoSCmY8yKpQgAAAAASUVORK5CYII=",
    "file_name": null
  }
}
